oag/src/main/java/org/owasp/oag/exception/ApplicationException.java

```diff
@@ -0,0 +1,32 @@
+package org.owasp.oag.exception;
+
+/**
+ * Usage: Throw this exception whenever the code is fine but the client provided bad data or the semantics of the call are not valid.
+ * User Reaction: Fix input and try again.
+ * Responsible to fix: Caller/Client.
+ * Log level: Usually info, max Warning.
+ * May create subclasses do distinguish errors and provide more details.
+ */
+public class ApplicationException extends AbstractException {
```

Comment:
I really like this approach with different exception classes for different kind of errors. ❤️

Since the ApplicationException class is used for all client related error we should think about adding the following annotation to exception class to make sure that we return appropriate response status: `@ResponseStatus(code = HttpStatus.BAD_REQUEST)`

Also, could we rename it to something like ClientException which is more intuitive for errors caused by the client.